tests/ci/benchmark_framework/compare_results.py

```diff
@@ -0,0 +1,63 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import pandas as pd
+import numpy as np
+
+if len(sys.argv) != 4:
+    print("Usage: compare_results.py [file1] [file2] [output filename]", file=sys.stderr)
+    sys.exit(1)
+
+file1 = sys.argv[1]
+file2 = sys.argv[2]
+
+if not (file1.endswith(".json") or file1.endswith(".csv")) and not (file2.endswith(".json") or file2.endswith(".csv")):
+    print("Provided files must either be .json files or .csv files", file=sys.stderr)
+    sys.exit(1)
+
+# read contents of file1 into a dataframe in preparation for comparison
+if file1.endswith(".json"):
+    df1 = pd.read_json(file1)
+else:
+    # this assumes we're using a csv generated by convert_json_to_csv.py
+    df1 = pd.read_csv(file1, skiprows=1, index_col=0)
+
+# read contents of file2 into a dataframe in preparation for comparison
+if file2.endswith(".json"):
+    df2 = pd.read_json(file2)
+else:
+    # this assumes we're using a csv generated by convert_json_to_csv.py
+    df2 = pd.read_csv(file2, skiprows=1, index_col=0)
+
+# only compare benchmarks that appear in both of the files
+if df1.shape[0] > df2.shape[0]:
+    df1 = df1[df1.iloc[:, 0].isin(df2.iloc[:, 0])]
+elif df2.shape[0] > df1.shape[0]:
+    df2 = df2[df2.iloc[:, 0].isin(df1.iloc[:, 0])]
```

Comment:
Currently this happens when the OpenSSL benchmark output produced by the speed tool is compared to BoringSSL or AWS-LC, since some tests in the speed tool are not compatible with OpenSSL so they are disabled. I will add a API doc link and explain the magic 0 and .shape :)

Reply:
Also add above explanation?